From 7206d5a146c9d55ebb9094cd0622519d38f73a0b Mon Sep 17 00:00:00 2001
From: Balamurugan Ramar <quic_bramar@quicinc.com>
Date: Tue, 22 Oct 2024 20:29:15 +0530
Subject: [PATCH] mac80211: Allow reconfig to complete during SSR mode0

During ssr mode0 + wifi down up scenario, if wifi down
completes first, mac80211 reconfig doesn't allow to
clearing FLAG RECOVERY bit. Add changes to allow reconfig
to clear the FLAG RECOVERY bit.

Signed-off-by: Balamurugan Ramar <quic_bramar@quicinc.com>
---
 net/mac80211/util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/mac80211/util.c b/net/mac80211/util.c
index 5dc7cdf..8754521 100644
--- a/net/mac80211/util.c
+++ b/net/mac80211/util.c
@@ -3017,7 +3017,7 @@ int ieee80211_reconfig(struct ieee80211_local *local)
 	 * If this is for hw restart things are still running.
 	 * We may want to change that later, however.
 	 */
-	if (local->open_count && (!suspended || reconfig_due_to_wowlan))
+	if (!suspended || reconfig_due_to_wowlan)
 		drv_reconfig_complete(local, IEEE80211_RECONFIG_TYPE_RESTART);
 
 	if (local->in_reconfig) {
-- 
2.34.1

